src/formula/callable_objects.hpp

```diff
@@ -25,10 +25,34 @@ class team;
 class terrain_type;
 class unit;
 class unit_type;
+namespace game_events {struct queued_event;}
 
 namespace wfl
 {
 
+// An object representing the state of the game, providing access to the map and basic information
+class gamestate_callable : public formula_callable
+{
+public:
+	// TODO: Constructor
+	// Currently it accesses all its state through the resources namespace, so nothing is passed into it.
+	
+	void get_inputs(formula_input_vector& inputs) const override;
+	variant get_value(const std::string& key) const override;
+};
+
+// An object representing the state of the current event; equivalent to Lua's wesnoth.current.event_context
```

Comment:
Doxygen comment.

Reply:
I was kinda unsure what to do here since it's more of a WFL API than a C++ API, but I can make it doxygen if you prefer.

Reply:
My thought is that even if it's implementing WFL, it's still C++ code that needs to be understood by anyone who needs to understand it later.